### DISCUSSION TITLE
mDNS lookups?

### DISCUSSION THREAD

**Discussion by mikronauts** [micropython/micropython#15791] [ESP8266]
I have ubuntu server on a Raspberry Pi, and I have enabled mdns. Computers on my network find 'home.local' without issues.
On an ESP8266 (latest MicroPython) no dice.

>>> usocket.getaddrinfo("home.local",5005,0,socket.SOCK_DGRAM)
Warning: unsupported getaddrinfo constraints
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OSError: -2

Same getaddrinfo works fine on external web server names.

I found https://forum.micropython.org/viewtopic.php?t=11498 which shows the same issue has been around for quite some time.

I have two potential workarounds, but would prefer to be able to do a simple getaddrinfo()

Workaround #1: set a static ip on my router
Workaround #2: scan my local network for my pi

Basically the pi will collect data from  ESP8266's and ESP32's and present via rrdtool

**Comment by bidrohini** [comment]
You can switch to ESP32. The ESP32 has better networking capabilities, including improved support for mDNS. If you can switch to using ESP32 instead of ESP8266, you might find that usocket.getaddrinfo() works as expected for .local addresses.

**Reply by mikronauts** [reply]
Thank you, will try that this morning.

I was trying to use some of my pile of esp8266's, but can use esp32's

**Reply by mikronauts** [reply]
Thank you - worked fine!
